Kitodo/src/main/java/org/kitodo/production/security/AESUtil.java

```diff
@@ -0,0 +1,160 @@
+
+    /**
+     * Encrypts a value by the secret using AES/CBC/PKCS5Padding algorithm.
+     *
+     * <p>
+     * Function generates salt to convert secret to AES-256 secret key and makes
+     * simple secrets more complex. In addition, function generates initialization
+     * vector (iv) to made encrypted value different when original value is the
+     * same. Salt and iv will be become one with the cipher text and result is
+     * converted to base64. As pseudocode the structure is as follows: BASE64( SALT(
+     * SALT_PREFIX + RANDOM ) + CIPHER TEXT + IV ( RANDOM ) )
+     * </p>
+     *
+     * @param value
+     *            The value to encrypt
+     * @param secret
+     *            The secret from config properties
+     * @return The encrypted value as base64 string.
+     */
+    public static String encrypt(String value, String secret)
+            throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidAlgorithmParameterException,
+            InvalidKeyException, BadPaddingException, IllegalBlockSizeException, InvalidKeySpecException {
+        // generate salt
+        byte[] salt = new byte[SALT_LENGTH];
+        new SecureRandom().nextBytes(salt);
+        System.arraycopy(SALT_PREFIX.getBytes(), 0, salt, 0, SALT_PREFIX.getBytes().length);
+
+        // generate iv
+        byte[] iv = new byte[IV_LENGTH];
+        new SecureRandom().nextBytes(iv);
+
+        Cipher cipher = Cipher.getInstance(ALGORITHM);
+        cipher.init(Cipher.ENCRYPT_MODE, getSecretKey(secret, salt), new IvParameterSpec(iv));
+        byte[] cipherText = cipher.doFinal(value.getBytes());
+
+        // prefix cipher text with salt and attach iv to the end
+        byte[] cipherCombined = new byte[SALT_LENGTH + cipherText.length + IV_LENGTH];
+
+        System.arraycopy(salt, 0, cipherCombined, 0, SALT_LENGTH);
+        System.arraycopy(cipherText, 0, cipherCombined, SALT_LENGTH, cipherText.length);
+        System.arraycopy(iv, 0, cipherCombined, SALT_LENGTH + cipherText.length, iv.length);
+
+        return Base64.getEncoder().encodeToString(cipherCombined);
+    }
+
+    /**
+     * Checks if value is encrypted.
+     *
+     * <p>
+     * Function checks if value starts with defined salt prefix after base64
+     * decoding.
+     * </p>
+     *
+     * @param potentialEncryptedValue
+     *            The value to be checked.
+     * @return boolean true if value is encrypted using encrypt function
+     */
+    public static boolean isEncrypted(String potentialEncryptedValue) {
+        try {
+            byte[] cipherCombined = Base64.getDecoder().decode(potentialEncryptedValue);
+            byte[] saltPrefix = Arrays.copyOfRange(cipherCombined, 0, SALT_PREFIX.getBytes().length);
+            // check if cipher combined has salt prefix
+            return Arrays.equals(saltPrefix, SALT_PREFIX.getBytes());
+        } catch (IllegalArgumentException e) {
+            // not valid base64 string so
```

Comment:
Catching an exception without handling it is bad. At least log this exception in as trace or debug message.